app/helpers/application_helper/button/new_host_aggregate.rb

```diff
@@ -1,5 +1,13 @@
 class ApplicationHelper::Button::NewHostAggregate < ApplicationHelper::Button::ButtonNewDiscover
+  def supports_button_action?
+    @supported_button_action ||= begin
+      filtered_providers = Rbac::Filterer.filtered(ManageIQ::Providers::CloudManager)
+      filtered_providers.empty? ? false : filtered_providers.all? { |ems| ems.supports?(:create_host_aggregate) }
+    end
+  end
+
   def disabled?
-    super || ManageIQ::Providers::CloudManager.all.none? { |ems| ems.supports?(:create_host_aggregate) }
+    self[:title] = _("No cloud providers support creating host aggregates.") unless supports_button_action?
+    super || !supports_button_action?
```

Comment:
If you're using the `@error_message`, you can have `super || @error_message.present?` here.